### PR TITLE
Make Points construction with properties consistent with setting properties

### LINKED ISSUE #2755: Creating properties *after* creating a points layer results in an inconsistent state
## 🐛 Bug

When properties are added *after* a points layer has been created (rather than at creation time), the `.current_properties` attribute is not properly initialized, so switching the face color mode causes a KeyError in that dictionary.

## To Reproduce

```python
import napari
import numpy as np

viewer = napari.Viewer()
points = viewer.add_points(np.random.random((5, 2) * 512)
points.properties = {
    'cat': np.random.randint(0, 5, size=5),
    'cont': np.random.random(5),
}
points.face_color_mode = 'cycle'
```

which causes:

```pytb
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-8-0ffc0dfa8ad5> in <module>
----> 1 pts.face_color_mode = 'cycle'

~/projects/napari/napari/layers/points/points.py in face_color_mode(self, face_color_mode)
    880     @face_color_mode.setter
    881     def face_color_mode(self, face_color_mode):
--> 882         self._set_color_mode(face_color_mode, 'face')
    883 
    884     def _set_color_mode(

~/projects/napari/napari/layers/points/points.py in _set_color_mode(self, color_mode, attribute)
    913                         'values': self.properties[new_color_property],
    914                         'current_value': np.squeeze(
--> 915                             self.current_properties[new_color_property]
    916                         ),
    917                     }

KeyError: 'cat'
```

## Expected behavior

The above lines:

```python
points = viewer.add_points(np.random.random((5, 2) * 512)
points.properties = {
    'cat': np.random.randint(0, 5, size=5),
    'cont': np.random.random(5),
}
```

should be equivalent to:

```python
points = viewer.add_points(
    np.random.random((5, 2) * 512)
    properties = {
        'cat': np.random.randint(0, 5, size=5),
        'cont': np.random.random(5),
    },
)
```

In the latter case, no error is raised when setting `pts.face_color_mode = 'cycle'`. (Though there are other reasons this doesn't work well currently, but that is a different issue.)

### PR DESCRIPTION
**Pull request by andy-sweet**
# Description
This makes the behavior of constructing a Points layer with properties and setting those properties consistent. Specifically, when the properties are set and thus clobbered, we now update the current properties in the same way the constructor does.

Even though the change is small, I marked it as WIP for a few reasons.
- I'm also working on the larger issue #2866, which may make this PR redundant.
- I'm unsure if the test represents a clear/common use of properties.
- I'm unsure how to avoid the warning in the test usage through the Points public API, and if that's something we should think about here.

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
Closes #2755.

# How has this been tested?
- [x] All the tests in test_points.py continue to pass.
- [x] A new test that reproduces the buggy behavior also passes.

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: napari/layers/points/_tests/test_points.py [hunk 1]
```diff
@@ -1593,3 +1594,29 @@ def test_prepare_properties():
...
+    rng = np.random.default_rng(0)
+    num_points = 3
+    points = Points(rng.random((num_points, 2)))
+
+    points.properties = {
+        'cat': rng.integers(low=0, high=num_points, size=num_points),
+        'cont': rng.random(num_points),
+    }
+
+    with pytest.warns(UserWarning):
```
Comment (jni) [inline napari/layers/points/_tests/test_points.py:1613]: What's the warning here? Could you maybe add a comment about it?

**Review by jni — APPROVED** [review]
@andy-sweet this LGTM, I left a completely minor comment. I don't think this work is redundant, it improves things and adds a test that should be satisfied in subsequent work. So 👍 from me to merge this.

**Review by sofroniewn — APPROVED** [review]
I'm also +1 to merging this as a fix for other issue, even if larger improvements are still to come. Maybe just add a comment on the user warning and then this is good to go. Thanks!!

File: napari/layers/points/_tests/test_points.py (hunk 1 above)
Comment (andy-sweet) [reply napari/layers/points/_tests/test_points.py:1613]: Done.

The presence of this warning also makes me think this test doesn't really represent an intended use. @kevinyamauchi : if you have any opinions here or any suggestions to make this test closer to a use of setting `face_color_mode`, that would be great!

**Comment by andy-sweet** [comment]
Thanks for the reviews. Agreed that there is some use to merging this fix alone, especially with the test, though it would be nice to get a little more confirmation from @kevinyamauchi that the test is exercising a meaningful behavior.

**Pull request merged**